coala_quickstart/info_extractors/GruntfileInfoExtractor.py

```diff
@@ -0,0 +1,265 @@
+    supported_file_globs = ("Gruntfile.js",)
+    supported_info_kinds = (LintTaskInfo, MentionedTasksInfo)
+
+    def parse_file(self, fname, file_content):
+        js_parser = PyJsParser()
+        return js_parser.parse(file_content)
+
+    def find_information(self, fname, parsed_file):
+        results = []
+
+        npm_tasks = self.get_npm_tasks(parsed_file)
+        linters = self.extract_lint_subtasks(parsed_file)
+        config = self.get_configurations(parsed_file, linters)
+
+        if npm_tasks:
+            results.append(
+                MentionedTasksInfo(fname, npm_tasks))
+
+        for linter in linters[:]:
+            include_paths_info = None
+            ignore_paths_info = None
+            if config.get(linter):
+                linter_config = config[linter]
+                globs = self.extract_globs(linter_config)
+                linter_config = self.get_task_config(linter_config)
+                if globs.get("include"):
+                    include_paths = globs["include"]
+                    include_paths_info = IncludePathsInfo(
+                        fname, include_paths) if include_paths else None
+                if globs.get("ignore"):
+                    ignore_paths = globs["ignore"]
+                    ignore_paths_info = IgnorePathsInfo(
+                        fname, ignore_paths) if ignore_paths else None
+            results.append(
+                LintTaskInfo(fname,
+                             linter,
+                             None,
+                             include_paths_info,
+                             ignore_paths_info,
+                             linter_config))
+
+        return results
+
+    def extract_lint_subtasks(self, parsed_file):
+        """
+        Extract the lint subtasks from the parsed JS file.
+        Looks for identifiers like:
+        ``grunt.registerTask( 'lint', [ 'csslint', 'jshint' ] );``
+
+        :param parsed_file:
+            An instance of PyJsParser().parse
+        :return:
+            A list of lint subtasks
+        """
+        # task names to match
+        keys_to_match = ['lint']
+
+        # Serch for grunt.registerTask() identifiers
+        search_results = search_object_recursively(parsed_file, "callee", {
+            "computed": False,
+            "type": "MemberExpression",
+            "property": {
+                "name": "registerTask",
+                "type": "Identifier"
+                },
+            "object": {
+                "name": "grunt",
+                "type": "Identifier"
+                }
+            })
+
+        lint_subtasks = []
+        is_lint = False
+
+        for res in search_results:
+            arguments = res["object"].get('arguments')
+            # check if the task is a lint sub-task
+            if arguments:
+                for arg in arguments:
+                    if arg.get('value') in keys_to_match:
+                        is_lint = True
+                        break
```

Comment:
Too much cyclomatic complexity in here. I'm pretty sure this can be achieved in a much simpler way.